src/snowflake/snowpark/dataframe.py

```diff
@@ -584,15 +584,9 @@ def _set_ast_ref(self, sp_dataframe_expr_builder: Any) -> None:
         """
         Given a field builder expression of the AST type SpDataframeExpr, points the builder to reference this dataframe.
         """
-        # TODO: remove the None guard below once we generate the correct AST.
-        if self._ast_id is None:
-            if FAIL_ON_MISSING_AST:
-                _logger.debug(self._explain_string())
-                raise NotImplementedError(
-                    f"DataFrame with API usage {self._plan.api_calls} is missing complete AST logging."
-                )
-        else:
-            sp_dataframe_expr_builder.sp_dataframe_ref.id.bitfield1 = self._ast_id
+        # TODO: remove once we generate the correct AST.
```

Comment:
nit: I think we can rename to `check_missing_ast` - no need for the debug prefix :)

Reply:
I prefer the debug in the name to indicate this isn't meant to be a production check. It's a fairly common practice (e.g. https://support.microsoft.com/en-us/topic/fix-assert-failure-of-the-debugchk-macro-when-you-delete-a-wi-fi-profile-in-a-debug-build-of-windows-embedded-compact-7-aa89c56d-ea8b-74f4-bd1b-751bcffe3dbb).